redisson/src/main/java/org/redisson/RedissonBaseLock.java

```diff
@@ -73,11 +74,16 @@ public synchronized void addThreadId(long threadId) {
         public synchronized boolean hasNoThreads() {
             return threadIds.isEmpty();
         }
-        public synchronized Long getFirstThreadId() {
+        public synchronized Long getLastThreadId() {
```

Comment:
I didn't see any reason for this to get FirstThread, since new threads are added at the end of map, so it makes more sense to me to return the last ID. This doesn't have any impact on my fix by the way.

Reply:
we need to return firstThreadId since it's the current lock owner.

Reply:
I will change this, but I do understand that that last thread Id was the owner of this lock.

Reply:
for read/write lock multiple read locks can hold it.